### PR TITLE
GH: Add Contribution Guidelines and Pull Request template

### PR DESCRIPTION
**Pull request by MotoLegacy**
### Description of Changes
---
This PR adds a `CONTRIBUTING.md` file as well as a Pull Request template (which you're seeing now!) in an attempt to better organize our Git history to be more welcoming to developers and keep things a little more tidy. Please review and give lots of feedback so that we can make sure this change is perfect!

### Visual Sample
---
[Contribution File](https://github.com/lavenderdotpet/LibreQuake/blob/cypress/contributing-guidelines/CONTRIBUTING.md)
[Pull Request Template](https://github.com/lavenderdotpet/LibreQuake/blob/cypress/contributing-guidelines/.github/pull_request_template.md)

### Checklist
---

- [x] I have read the LibreQuake contribution guidelines
- [x] I have thoroughly tested my changes to the best of my ability
- [x] I confirm I have not contributed anything that would impact LibreQuake's licensing and usage
- [ ] This Pull Request fixes a **critical** issue that should be reviewed and merged as soon as possible

### CODE REVIEW

**Comment by MotoLegacy** [comment]
I forgot to mention, we should have something like a two-week adjustment period when/if this is merged to give people time to adapt to this change and not severely impact their ability to contribute :)

File: CONTRIBUTING.md
Comment (MotoLegacy) [inline CONTRIBUTING.md:60]: Typo spotted!

**Comment by Nolcoz** [comment]
After reading all changes, they all seem pretty well tought out.

My only questions are more so with existing branches and to fully understand how branches should be like.

-------
**FIRST QUESTION**
So, from reading how branches should be I got two interpretations of how they work and want to know which one is the correct one.
- First one:
With the new system, I should make a new branch with a name reflecting a new thing being added/changed.
Like for example, if I make new textures for the keys, I should make a new branch named "nolcoz/key-textures", and if that gets merged with the main repo, that branch should stop being used after that. Any new changes to key textures should be a new branch.

- Second one:
Same as the first one BUT the branch can still be used after the intial merge, as long as it has to do with changes related to the name of the branch. In this example being related to changing key textures

------
**SECOND QUESTION**
The other question I have is more so with my specific case.
With a branch like mine, which is simply named Nolcoz, I used it to add any sort of changes with no real overaching theme.
After this, should branches like that stop being used?

**Comment by MotoLegacy** [comment]
Hi Nolcoz,

1. The first one. Branches should be created for a specific purpose, and once that purpose has been satisfied in a merged Pull Request, it should be retired. This follows standard practice, so we'd be doing away with the whole "megabranch" thing that's been done previously that adds a lot of redundant commits and merge conflicts and upstream pulls etc.

2. Yes, that practice should be retired in favor of this more specific principle outlined in the contribution guide. Think of branches literally -- like branches on a tree. They should be allowed to "die" (stop growing) and should not wrap around becoming discombobulated and going in and out of a tree's trunk.

**Review by Nolcoz — APPROVED** [review]
Okay, thanks for helping me with that!

I think its good for aproval then.

**Review by lavenderdotpet — DISMISSED** [review]
looks good to me

**Comment by lavenderdotpet** [comment]
do u wish for me to wait for others to reply

**Comment by MotoLegacy** [comment]
> do u wish for me to wait for others to reply

I would encourage you to read thoroughly and make suggestions where needed. Regardless, we need to wait for as much input as possible because of the scope of this change.

**Comment by lavenderdotpet** [comment]
it looks good to me but I'll wait till I have a  chance to thoroughly thoroughly read it  and that time is not now

**Comment by MotoLegacy** [comment]
> it looks good to me but I'll wait till I have a chance to thoroughly thoroughly read it and that time is not now

Understood. Can you please revoke your approval then?

File: .github/pull_request_template.md
Comment (pnahratow) [inline .github/pull_request_template.md:11]: Might need a hint that you are supposed to put capital X in the brackets to make the check work

File: .github/pull_request_template.md
Comment (pnahratow) [inline .github/pull_request_template.md:3]: I believe you can use xml comments in markdown `<!-- text here -->` this way you could place the hints here but the user would need to replace the text and could just leave it. Not sure though.

File: .github/pull_request_template.md
Comment (pnahratow) [inline .github/pull_request_template.md:13]: This one is kind of meh. Yeah sure but the Visual Sample thing kind of takes care of this already (almost) and this is just a reminder that I have to check everytime but what does it really mean then. WIll I go "Ah, you know lets stop right here and do some testing before I open this PR". I kinda doubt it.

File: CONTRIBUTING.md
Comment (pnahratow) [inline CONTRIBUTING.md:76]: I don't like this. I use `git commit -m ...` most of the time and only extend to using the body when there is explaining to do. However I'm not adamant about this. Just asking to reconsider.

File: CONTRIBUTING.md
Comment (pnahratow) [inline CONTRIBUTING.md:103]:
```suggestion
If the change fixes a critical issue (such as a crash, broken animation, etc.) an exception should be made for the 48 hour rule, **however all other prerequisites must be met!**
```

**Review by pnahratow — COMMENTED** [review]
I like your writing style. This was all very clearly communicated.

File: .github/pull_request_template.md
Comment (MotoLegacy) [reply .github/pull_request_template.md:11]: I thought about this but evaluated that teaching markdown is out of scope, and you're also able to check the boxes in the preview and after submitting the PR. Can change if you think this is still necessary considering.

File: .github/pull_request_template.md
Comment (MotoLegacy) [reply .github/pull_request_template.md:3]: Yes, you can use the comment syntax there. Will go ahead and revise, thanks for the feedback.

File: .github/pull_request_template.md
Comment (MotoLegacy) [reply .github/pull_request_template.md:13]: My thought process when opting to add this was a prediction that many contributors will see something work in their very specific case (such as a test map) and assume things are working correctly. I wanted the act of checking this box to be a conscious decision, and acknowledgement that we take finding regressions seriously.

I agree if you are like me (for better or for worse) you will go "this check is a waste of my time" but considering the fact we have a lot of artists who come-in-and-get-out, we might actually be able to encourage them to spend a little more time verifying their changes.

Curious about your thoughts, happy to revise if you disagree.

File: CONTRIBUTING.md
Comment (MotoLegacy) [reply CONTRIBUTING.md:76]: Upon further consideration I agree, however I do think we need some standard to dictate that you should add a commit body given some factors (such as size/impact). Hard to come up with something that isn't super arbitrary though. What do you think?

File: CONTRIBUTING.md
Comment (MotoLegacy) [inline CONTRIBUTING.md:60]:
```suggestion
Additionally, as this repository contains multiple diverse components, we ask that you prefix your commit message pertaining to the component you are modifying. Here are acceptable prefixes to use:
```

File: CONTRIBUTING.md
Comment (pnahratow) [reply CONTRIBUTING.md:76]: I'd say just skip it. https://cbea.ms/git-commit/ is also super vague here. 

"More detailed explanatory text, if necessary."

File: .github/pull_request_template.md
Comment (pnahratow) [reply .github/pull_request_template.md:11]: No its okay

File: .github/pull_request_template.md
Comment (pnahratow) [reply .github/pull_request_template.md:13]: We can leave it in. Also I didn't know that you can click to check the boxes so that actually becomes less tedious.

**Comment by MotoLegacy** [comment]
Thank you for your thorough review and feedback @pnahratow !

**Review by pnahratow — APPROVED** [review]

File: CONTRIBUTING.md
Comment (pnahratow) [inline CONTRIBUTING.md:70]: I would like to have this list accessible when creating new PR.

File: CONTRIBUTING.md
Comment (pnahratow) [inline CONTRIBUTING.md:101]: I see the intention here but I disagree with adding a 48h delay for everything but critical issues. Sometimes development just wants to move forward. If there are 4 approvals on a PR and people want something to go in it should be able to. Like if we could move "We really want it" under the banner of "critical" then I'm okay with this.

Comment (MotoLegacy) [reply CONTRIBUTING.md:101]: Would you be alright if this decreased to 24 hours? We have had several occurrences wherein we have rushed PRs, with multiple approvals, and paid the price later where we realize there was a case we forgot to address (I have been one to make these sorts of mistakes as well)

Some of us work full time and that can be detrimental to our ability to review a PR as soon as it is opened. My worry is that without this getting we will be less able to catch issues before they make it into prod because I or someone else wasnt around in that 6 hour window, or whatever it may be.

Comment (pnahratow) [reply CONTRIBUTING.md:101]: Yeah I think I would like 24h better. As I said I do understand the concern.

Comment (lavenderdotpet) [reply CONTRIBUTING.md:101]: maybe 48 with 2 reviews and 24 for 3 reviews?

File: CONTRIBUTING.md
Comment (lavenderdotpet) [reply CONTRIBUTING.md:70]: also same would want this 
can hide it within this
```<!-- This content will not appear in the rendered Markdown -->```

**Review by pnahratow — APPROVED** [review]

**Comment by lavenderdotpet** [comment]
> I forgot to mention, we should have something like a two-week adjustment period when/if this is merged to give people time to adapt to this change and not severely impact their ability to contribute :)

also branches that currently already exist should get grandfathered in (as in if a branch for a soon to exist pull request existed before this got merged the one running the branch shouldn't have to rename it and just follow everything else other then the branch name to the best of their ability)

**Review by lavenderdotpet — CHANGES_REQUESTED** [review]
### Description of Changes
---

### Visual Sample
---

### Checklist
---

### i think this ↑ should be changed to this ↓


## Description of Changes

## Visual Sample

## Checklist

 bottom either using 1 or 2 hashes " # "
1 or 2 hashes on github markdown have a built in title line rather then using the " --- "

this is just personal pref since i often use triple hash for text in my pull requests so using triple as the title lines feels a bit off for me

**Comment by MotoLegacy** [comment]
Hi @lavenderdotpet, my main concern is that I personally find the second option harder to read since the division lines are smaller (this is especially a problem on mobile devices). Can you reconsider? If not I will meet the request anyway.

**Review by lavenderdotpet — APPROVED** [review]

**Comment by lavenderdotpet** [comment]
> Hi @lavenderdotpet, my main concern is that I personally find the second option harder to read since the division lines are smaller (this is especially a problem on mobile devices). Can you reconsider? If not I will meet the request anyway.

alrighty 
ready to merge

**Pull request merged**
